solves issue #11

stankasvecova opened the pull request:
1. simplified signUp & signIn registration .html & .css
2. signupsignin.js handles firebase functions such as 'create user', 'sign-in with email & password', 'show status of user', then shows a block of code when signed-in. Attempted to add 'sign in with pop-up' from google as a provider. 

Did not find how to do 'sign-out' though. 
It's a start. Shoot what's wrong, what to change. Thanks!

stankasvecova committed:
solves issue #11

1. simplified signUp & signIn registration .html & .css
2. signupsignin.js handles firebase functions such as 'create user', 'sign-in with email & password', 'show status of user', then shows a block of code when signed-in. Attempted to add 'sign in with pop-up' from google as a provider. 

Did not find how to do 'sign-out' though. 
It's a start. Shoot what's wrong, what to change. Thanks!

MrTim commented on signupsignin.css:
```diff
@@ -0,0 +1,231 @@
+    background-color: #fff;
+    border-color: orangered
+}
+
```
please remove all those empty lines

stankasvecova replied:
Done

MrTim commented on signupsignin.html:
```diff
@@ -0,0 +1,57 @@
+
+<body>
+    <script src="signupsignin.js"></script>
+    <script src="/__/firebase/7.15.1/firebase-app.js"></script>
```
hm...where those paths are pointing to? If you would open a developer tool in your browser you would see a lot of FILE_NOT_FOUND errors because browser trying to find them in your project. But you should take them from a server like that:
```js
    <script src="https://www.gstatic.com/firebasejs/7.15.1/firebase-app.js"></script>
    <script src="https://www.gstatic.com/firebasejs/7.15.1/firebase-analytics.js"></script>
    <script src="https://www.gstatic.com/firebasejs/7.15.0/firebase-auth.js"></script>
    <script src="https://www.gstatic.com/firebasejs/7.15.0/firebase-firestore.js"></script>
```

MrTim replied:
More on top: 
I think you forgot what I mentioned that you need to use 'setup with CDN' - [setup](https://firebase.google.com/docs/web/setup#from-the-cdn)

so what you need to do is to use this:
```html
    <script src="https://www.gstatic.com/firebasejs/7.15.1/firebase-app.js"></script>
    <script src="https://www.gstatic.com/firebasejs/7.15.1/firebase-analytics.js"></script>
    <script src="https://www.gstatic.com/firebasejs/7.15.0/firebase-auth.js"></script>
    <script src="https://www.gstatic.com/firebasejs/7.15.0/firebase-firestore.js"></script>
    <script>
        const firebaseConfig = {
            apiKey: "AIzaSyBbuvg-wCgz30zyD4zETU0HiHiC5_FulWc",
            authDomain: "good-teens.firebaseapp.com",
            databaseURL: "https://good-teens.firebaseio.com",
            projectId: "good-teens",
            storageBucket: "good-teens.appspot.com",
            messagingSenderId: "260739514487",
            appId: "1:260739514487:web:8197a36e539f090f2ec672",
            measurementId: "G-0WBRPFKT5Z"
        };

        // Initialize Firebase
        firebase.initializeApp(firebaseConfig);
    </script>
```

and after that you would have your application ready to work with firebase.

MrTim replied:
and one more thing: 
in firebaseConfig - your `appId` looks strange you can double check it in your fireBase webpage.

stankasvecova replied:
Yes, I did 'setup with CDN' according to the page. I will do it now again to see if anything changes. 
appID: 1:260739514487:web:8197a36e539f090f2ec672 - i know, looks weird.
![obrázok](https://user-images.githubusercontent.com/62797218/84770292-90c6d280-afd7-11ea-8c5a-4f78283fdf9f.png)

MrTim commented on signupsignin.html:
```diff
@@ -0,0 +1,57 @@
+<!DOCTYPE html>
```
this file should go into html folder.

stankasvecova replied:
will do

MrTim commented on signupsignin.js:
```diff
@@ -0,0 +1,130 @@
+
```
this file should got into js forlder.

stankasvecova replied:
will do

MrTim commented on signupsignin.js:
```diff
@@ -0,0 +1,130 @@
+
+const firebaseConfig = {
```
you need to remove this config from here and use it as i wrote above.

stankasvecova replied:
done

MrTim commented on signupsignin.js:
```diff
@@ -0,0 +1,130 @@
+  // ...
+});
+
+signUpEl.addEventListener(click, signUp);
```
first argument of `addEventListener` is a string but right now it's a variable `click` and browser will argue about it.
```js
  signUpEl.addEventListener('click', signUp);
```

stankasvecova replied:
oh yes, thanks.

MrTim reviewed (changes requested):
I added a comment about why your firebase setup is not working and couple more comments. 
You can fix those and see what you would have after that and continue.

stankasvecova committed:
updated debugged signup-signin files

MrTim commented on html/signupsignin.html:
```diff
@@ -8,16 +8,10 @@
     <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
     <link rel="stylesheet" href="signupsignin.css">
```
you did move files but you forgot to change the import path, so at the moment all styles should be broken.

stankasvecova replied:
Connected them now

MrTim commented on html/signupsignin.html:
```diff
@@ -52,6 +46,26 @@ <h2>Sign in</h2>
         </div>
     </div>
+
+    <script src="signupsignin.js"></script>
```
the same here, this file would not be found under this location because you moved it to js forlder.

stankasvecova replied:
Connected them now.

MrTim commented on html/signupsignin.html:
```diff
@@ -52,6 +46,26 @@ <h2>Sign in</h2>
     </div>
+
+    <script src="signupsignin.js"></script>
+    <script src="https://www.gstatic.com/firebasejs/7.15.1/firebase-app.js"></script>
```
according to the documentation, this imports should be before you'll use them. So please move them up, together with firebaseConfig and initialization.

stankasvecova replied:
Moved them up.
However I tested it and it's still not adding users into console in firebase.
Chrome console is not giving me any bugs either, so a bit lost...

MrTim reviewed (changes requested):
Commented what still needs to be fixed. After that you would be able to register user with email and password.

stankasvecova committed:
updates

stankasvecova committed:
part of script moved up, part of script moved to the bottom

MrTim reviewed (dismissed):
Ok seems like it's working now! 
Let's merge it into master.

stankasvecova committed:
Added signout function

BTBTravis commented:
This PR is getting pretty big probably should have merged at 31552ac and handle other changes in a follow up PR.

BTBTravis commented:
My observations:

1. Facebook and Google buttons don't seem to do anything, perhaps this is planned for future PR
1. Signout button shows even when I've never signed in. Should only show for logged in users no? Furthermore whats point of signout button on login page. If users are redirected after successful signup there is never a case where users in on login page and already logged in right?
1. I can create account without agreeing to terms of service
1. when I create an account there is no feedback to let user know they are logged in. perhaps there should be redirect to page that just says logged in for now or something. 
1. there is debbugger statement left in the code on signout causing page to hang.

BTBTravis reviewed (commented):
Added my observations

stankasvecova commented:
Hey Travis and all,

Thanks for the comments, answers are here: 
_Facebook and Google buttons_ => yes, upcoming PR.
_Signout button shows even when I've never signed in_ => yes that was work in progress. Now moved to after signed-in page. 
_I can create account without agreeing to terms of service_ => in html form, I have it as 'required' but it still lets me sign up even when unticked. Not sure how to solve it yet. 
_when I create an account there is no feedback to let user know they are logged in_ => yes work in progress.
_there is debbugger statement left in the code on signout causing page to hang_ => will remove, thanks.

stankasvecova committed:
updates

frustrating as it seems to work with one test and then doesn't with the next. SignUp, signIn, signOut and status change is in there. External providers login is not yet set-up.

Very grateful if anyone sees why it seems working one minute (network gave me 200 for everything), then i do it again and it just doesn't even go into the function). I get that the 'else' part in signedIn function can be left out, but it didn't work even if I left it out.

MrTim commented on html/signupsignin.html:
```diff
@@ -9,6 +9,7 @@
     <link rel="stylesheet" href="../css/signupsignin.css">
     <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/all.css"
         integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf" crossorigin="anonymous">
+    <link rel="stylesheet" href="createprofile.css">
```
there is no such file found. So ether remove this link or add the css file. And css files should be stored in css folder (as a reminder)

MrTim commented on js/signupsignin.js:
```diff
@@ -28,124 +27,124 @@ function signIn() {
+    var providerData = user.providerData;
+    signedInPage(user);
+  } else {
+    signIn()
```
This is not what you want to have. You want to display a signin html page but not execute signin function.

MrTim reviewed (changes requested):
Please adjust regarding comments.

stankasvecova committed:
Changed signup-signin-signout flow

According to MrTim's suggestions.

Both .html are hooked to setupFirebase.js.

According to status (signedin/ not signedin), user is redirected either to signupsignin.html or to personalProfile.html.

And yet, seems not working.

.JPG and .PNG will be removed to a respective folder when it gets created (will do a pr now).

MrTim commented on js/personalProfile.js:
```diff
@@ -0,0 +1,61 @@
+  firebase.auth().signOut().then(function() {
+    // Sign-out successful.
+    }).catch(function(error) {
+    // An error happened.
```
What would happen in case of error?

stankasvecova replied:
Added a little console message:
`function signOut() {
  firebase.auth().signOut().then(function() {}).catch(function(error) {
   console.error('Sign-out Error', error)
    });
}`

MrTim commented on html/signupsignin.html:
```diff
@@ -9,28 +9,14 @@
+    <script src="../js/setupFirebase.js"></script>
     <div class="container">
         <div class="column1">
             <form action="#">
```
this action is breaking the proper execution of signup (maybe signin as well). Please remove it.

stankasvecova replied:
I removed the 'action' from the <form> completely, but it didn't help. It isn't yet working, a bug must be somewhere else. I will keep trying, will make another push once it works.

MrTim reviewed (changes requested):
Couple things to fix.

stankasvecova committed:
updates

stankasvecova committed:
<form> has now EventListener to prevent default

But the site started looping endlessly between signupsignin.html and personalProfile.html.

I've tried various ways how to stop it looping but so far nothing worked. Help?

stankasvecova committed:
updates

stankasvecova committed:
looping between two files

stankasvecova committed:
updates

Sing-in and Sign-out seems still not working

stankasvecova committed:
final tweaks

MrTim commented on html/personalProfile.html:
```diff
@@ -16,12 +16,12 @@
     <script src="../js/setupFirebase.js"></script>
     <header>
-        <h1 class="regTitle"><img src="logo1.png" class="gtlogo" alt="goodteens"></h1>
+        <h1 class="regTitle"><img src="../docs/logo1.png" class="gtlogo" alt="goodteens"></h1>
```
normally all icons and imgs are stored in `assets` forlder.

stankasvecova replied:
Will do for the next PR. It was misunderstanding - i thought no other folders were allowed to have created, so I put it there. I will know to simply create an asset folder.

MrTim commented on js/setupFirebase.js:
```diff
@@ -1,33 +1,29 @@
\ No newline at end of file
+  if (user) {
+    // User is signed in.
+    var displayName = user.displayName;
```
this variable are not used. If code is not used please delete it - this is kinda a rule.

stankasvecova replied:
deleted

MrTim reviewed (dismissed):
Please merge it.

stankasvecova committed:
deleted unused variable

MrTim commented on js/setupFirebase.js:
```diff
@@ -15,7 +15,6 @@ firebase.analytics();
   if (user) {
     // User is signed in.
-    var displayName = user.displayName;
     var email = user.email;
```
actually comment was not only for displayName but for all variables here. You're not using them.

stankasvecova committed:
getting rid of redundant variables

stankasvecova committed:
Merge branch 'signup-signin' of https://github.com/redi-js-teachers/js_sprint_2020_final_project_stankasvecova into signup-signin

MrTim reviewed (approved):
I'll approve it even though google and facebook auth is not working so far.
Please work on it in next branch!
